openpype/hosts/standalonepublisher/plugins/publish/collect_editorial_csv.py

```diff
@@ -0,0 +1,752 @@
+        }
+        return filename, representation_data
+
+    def _get_row_value_with_validation(
+        self, column_name, row_data, column_config, default_value=None
+    ):
+        """Get row value with validation"""
+        # get column data from column config
+        column_data = column_config.get(column_name)
+        if not column_data:
+            raise KeyError(
+                f"Column '{column_name}' not found in column config."
+            )
+
+        # get column value from row
+        column_value = row_data.get(column_name)
+        # get column type
+        column_type = column_data["type"]
+        # get column validation regex
+        column_validation = column_data["validate"]
+        # get column default value
+        column_default = default_value or column_data["default"]
+
+        if column_type == "number" and column_default == 0:
+            column_default = None
+
+        # check if column value is not empty string
+        if column_value == "":
+            # set default value if column value is empty string
+            column_value = column_default
+
+        # set column value to correct type following column type
+        if column_type == "number" and column_value is not None:
+            column_value = int(column_value)
+        elif column_type == "bool":
+            column_value = column_value in ["true", "True"]
+
+        # check if column value matches validation regex
+        if (
+            column_value is not None and
+            not re.match(str(column_validation), str(column_value))
+        ):
+            raise ValueError(
+                f"Column '{column_name}' value '{column_value}' "
+                f"does not match validation regex '{column_validation}' \n"
+                f"Row data: {row_data} \n"
+                f"Column data: {column_data}"
+            )
+
+        return column_value
+
+    def _create_new_instance(
+        self, context, csv_file_path, passing_instance_data
+    ):
+        """Create new instance from the csv editorial file"""
+        dirname, filename = os.path.split(csv_file_path)
+        basename, _ = os.path.splitext(filename)
+
+        representation = {
+            "name": "csv",
+            "ext": "csv",
+            "files": filename,
+            "stagingDir": dirname,
+        }
+
+        # create new instance from the csv editorial file
+        new_instance = context.create_instance(
+            basename, family="editorial_csv",
+        )
+        # pass data from original instance to the new one
+        new_instance.data.update(passing_instance_data)
+
+        new_instance.data.update({
+            "label": f"CSV: {filename}",
+            "subset": (
+                passing_instance_data["subset"] + basename.replace("_", "")),
+            "representations": [representation],
+            "stagingDir": dirname,
+            "stagingDir_persistent": True,
+        })
+
+    def config_columns_data(self):
```

Comment:
I wonder whether if this were using `jsonschema` validations that it might be much easier to get very good error reports for?

Reply:
for reference look here https://github.com/ynput/OpenPype/pull/5795#discussion_r1440461055

Reply:
Did you link the wrong comment? The linked comment doesn't clarify much for me.

Reply:
I guess it really makes no sense to merge the `standalonepublisher` related changes. @jrsndl what do you thing? Once the Traypublisher will be working, nobody will use Standalone publisher, right?